src/io.cpp

```diff
@@ -618,45 +618,105 @@ GeneratedPipelineInput::GeneratedPipelineInput(const Catalog &catalog,
     }
 }
 
+
+Attitude CoverageTest(std::default_random_engine* pReng) { //CoverageTest() takes a Pointer as the paramter now
+
+
+    //TODO: check whether the random floats have the right precision (how many decimal points)
+
+
+
+    std::uniform_real_distribution<float> randomAngleDistribution(0, 1);
+
+    // normally the ranges of the Ra and Dec are:
+    // Dec: [-90 deg, 90 deg] --> [-pi/2 rad, pi/2 rad], where negative means south
+    // of the celestial equator and positive means north
+    // Ra: [0 deg, 360 deg] --> [0 rad, 2pi rad ]
+    // Roll: [0 rad, 2 pi rad]
+
+    float randomRa = 2 *  M_PI * randomAngleDistribution(*pReng);
+    float randomDec = (M_PI / 2) - acos(1 - 2 * randomAngleDistribution(*pReng)); //acos returns a float in range [0, pi]
+    float randomRoll = 2 *  M_PI * randomAngleDistribution(*pReng);
+
+    Attitude randAttitude = Attitude(SphericalToQuaternion((randomRa),
+                                                         (randomDec),
+                                                         (randomRoll)));
+
+    return randAttitude;
+}
+
 /// Create a GeneratedPipelineInput based on the command line options in `values`
 PipelineInputList GetGeneratedPipelineInput(const PipelineOptions &values) {
     // TODO: prompt for attitude, imagewidth, etc and then construct a GeneratedPipelineInput
 
+    // get time-based seed if coverage-test
+    int seed;
+
+    if (values.coverageTest != "") {
+
+        // ctime header file for this
+        seed = time(0);
+    } else {
+        seed = values.generateSeed;
+    }
+
+    std::default_random_engine reng(seed);
+
+    //Created Pointer for reng
+    //Code quality (where should you put the asterisk for the pointer)
```

Comment:
Code style depends on the project, but in LOST, the asterisk should be right-aligned (next to the variable name, not the type).

Also, in general, . Instead just use `&` whenever you need to dereference. For example just pass `&reng` to `CoverageTest` later.